week1/main.js

```diff
@@ -0,0 +1,174 @@
+    ".burger__card__detail__name"
+  );
+  if (currentBurger.innerText in addList) {
+    continueAdd(e, currentBurger);
+  } else {
+    initAdd(e, currentBurger);
+  }
+}
+
+function initAdd(e, currentBurger) {
+  addList[currentBurger.innerText] = [
+    1,
+    `₩${priceList[currentBurger.innerText]}`,
+  ];
+
+  const burgerCart = $(".cart__menu");
+  const menuContainer = document.createElement("div");
+  const menuText = document.createElement("div");
+  const countText = document.createElement("input");
+  const priceText = document.createElement("div");
+  const delBtn = document.createElement("button");
+
+  burgerCart.appendChild(menuContainer);
+  menuContainer.appendChild(menuText);
+  menuContainer.appendChild(countText);
+  menuContainer.appendChild(priceText);
+  menuContainer.appendChild(delBtn);
+
+  menuContainer.classList.add("add__menu");
+  menuText.classList.add("add__menu__text");
+  countText.classList.add("add__menu__count");
+  countText.id = `${currentBurger.innerText}`;
+  priceText.classList.add("add__menu__price");
+  delBtn.classList.add("delete__btn");
+
+  menuText.textContent = currentBurger.innerText;
+  countText.value = addList[currentBurger.innerText][0];
+  priceText.textContent = addList[currentBurger.innerText][1];
+  delBtn.textContent = "X";
+  countText.type = "number";
+
+  addAmount(currentBurger);
+  clickCancelBtn();
+  checkInput(currentBurger);
+}
+
+function checkInput(currentBurger) {
+  const addCount = document.getElementById(currentBurger.innerText);
+  addCount.addEventListener("click", (e) => {
+    const changedInput = Number(addCount.value);
+    amountPrice +=
+      (changedInput - addList[currentBurger.innerText][0]) *
+      priceList[currentBurger.innerText];
+    amountText.textContent = `₩${amountPrice.toLocaleString()}`;
+    addList[currentBurger.innerText][0] = changedInput;
+  });
+}
+
+function continueAdd(e, currentBurger) {
+  continueCount(currentBurger);
+  addAmount(currentBurger);
+}
+
+function continueCount(currentBurger) {
+  const addCount = document.getElementById(currentBurger.innerText);
+  addCount.value = Number(addCount.value) + 1;
+  addList[currentBurger.innerText][0] += 1;
+}
+
+function addAmount(currentBurger) {
+  amountPrice += priceList[currentBurger.innerText];
+  amountText.textContent = `₩${amountPrice.toLocaleString()}`;
+}
+
+function clickCancelBtn() {
+  const cancelBtn = document.querySelectorAll(".delete__btn");
+  cancelBtn.forEach((btn) => {
+    btn.addEventListener("click", (e) => {
+      cancelMenu(e);
+    });
+  });
+}
+
+function cancelMenu(e) {
+  const del = e.currentTarget.parentNode;
+  if (addList[del.firstChild.innerText]) {
+    amountPrice -=
+      priceList[del.firstChild.innerText] *
+      addList[del.firstChild.innerText][0];
+    amountText.textContent = `₩${amountPrice.toLocaleString()}`;
+    delete addList[del.firstChild.innerText];
+    del.remove();
+  }
+  return;
+}
+
```

Comment:
두 함수 자체가 같은 역할을 하고 있어서 합쳐줘도 될 것 같고, X를 누르면 해당 버거는 전부 사라져야하니까 parentNode를 삭제하고 그다음에 총금액을 계산하는 함수를 불러오면 될 것 같아!
```suggestion
function clickCancelBtn() {
  const cancelBtn = document.querySelectorAll(".delete__btn");
  cancelBtn.forEach((btn) => {
    btn.addEventListener("click", (e) => {
      const del = e.currentTarget.parentNode;
      del.remove();
      //총합 계산
  }
    });
  });
}
```